### PR TITLE
Add Topics from twreporter-react-listing-components

### PR DESCRIPTION
**Pull request by YuCJ**

### CODE REVIEW

File: src/containers/Topics.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        url={`${LINK_PREFIX.TOPICS}${_.get(item, 'slug')}`}
+      />
+    )
+    return _.map(topics, _buildTopicBox)
+  }
+
+  render() {
+    const { items, total, topicEntities, postEntities, topicListError, topicError } = this.props
+    const topics = camelizeKeys(denormalizeTopics(items, topicEntities, postEntities))
+    const topicsLength = topics.length
```
Comment (nickhsine) [inline src/containers/Topics.js]: `_.get(topics, 'length', 0)`

File: src/containers/Topics.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+          linkTo={formatPostLinkTo(slug, style)}
+          linkTarget={formatPostLinkTarget(style)}
+        />
+      )
+    }
+    return _.map(posts, _buildPostJSX)
+  }
+
+  _buildTopicBoxes(topics) {
+    const _buildTopicBox = (item, index='') => (
```
Comment (nickhsine) [inline src/containers/Topics.js]: `index` not a number?

File: src/containers/Topics.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
+import React, { Component } from 'react'
+import { TopicsComponents } from 'twreporter-react-listing-components'
+import twreporterRedux from 'twreporter-redux'
+import { connect } from 'react-redux'
+import styled from 'styled-components'
+import { date2yyyymmdd, formatPostLinkTo, formatPostLinkTarget } from '../utils'
```
Comment (nickhsine) [inline src/containers/Topics.js]: where are `formatPostLinkTo ` and  `formatPostLinkTarget ` defined?

File: src/containers/Topics.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+function mapStateToProps(state) {
+  const items = _.get(state, [ reduxStateFields.topicList, 'items' ], [])
+  const total = _.get(state, [ reduxStateFields.topicList, 'total' ])
+  const topicListError = _.get(state, [ reduxStateFields.topicList, 'error' ], null)
+  const topicError = _.get(state, [ reduxStateFields.selectedTopic, 'error' ], null)
+  const entities = _.get(state, reduxStateFields.entities, {})
+  const topicEntities = _.get(entities, reduxStateFields.topicsInEntities, {})
+  const postEntities = _.get(entities, reduxStateFields.postsInEntities, {})
```
Comment (nickhsine) [inline src/containers/Topics.js]: If you could, compose/format the topic items you need here.
And just pass the `items` into `Topics`.
Decoupling the rendering logic and data composing logic will make `Topics` more flexible.

File: src/containers/Topics.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+const N_OF_FIRSTPAGE = 5
+const N_PER_PAGE = 5
+
+const PageContainer = styled.div`
+  position: relative;
+  width: 100%;
+  padding: 30px 0 0 0;
+`
+
+class Topics extends Component {
```
Comment (nickhsine) [inline src/containers/Topics.js]: `PureComponent` ?

File: src/containers/Topics.js (hunk 3 above)
Comment (YuCJ) [inline src/containers/Topics.js]: they are in `utils/url-processor.js`

File: src/containers/Topics.js (hunk 5 above)
Comment (YuCJ) [inline src/containers/Topics.js]: Some of props from redux are objects, will shallow comparing always return that the props are different?

File: src/containers/Topics.js (hunk 4 above)
Comment (YuCJ) [inline src/containers/Topics.js]: :+1:

File: src/containers/Topics.js (hunk 2 above)
Comment (YuCJ) [inline src/containers/Topics.js]: 😭

**Pull request merged**